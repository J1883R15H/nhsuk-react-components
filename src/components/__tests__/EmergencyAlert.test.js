import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import EmergencyAlert from '../emergency-alert';

describe('EmergencyAlert', () => {
  it('title matches snapshot', () => {
    const shallowMount = shallow(
      <EmergencyAlert.Title>Title</EmergencyAlert.Title>
    );
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('message matches snapshot', () => {
    const shallowMount = shallow(
      <EmergencyAlert.Message>Message</EmergencyAlert.Message>
    );
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('link matches snapshot', () => {
    const shallowMount = shallow(
      <EmergencyAlert.Link href="#">Link</EmergencyAlert.Link>
    );
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('last updated matches snapshot', () => {
    const shallowMount = shallow(
      <EmergencyAlert.LastUpdated>
        Updated 23 mins ago
      </EmergencyAlert.LastUpdated>
    );
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('all matches snapshot', () => {
    const fullMount = mount(
      <EmergencyAlert>
        <EmergencyAlert.Title>National Flu Outbreak</EmergencyAlert.Title>
        <EmergencyAlert.Message>
          There has been a national flu outbreak.
          <EmergencyAlert.Link href="https://www.nhs.uk">
            How does it affect me
          </EmergencyAlert.Link>
        </EmergencyAlert.Message>
        <EmergencyAlert.LastUpdated>
          Updated 23 mins ago
        </EmergencyAlert.LastUpdated>
      </EmergencyAlert>
    );
    expect(fullMount).toMatchSnapshot();
    fullMount.unmount();
  });
  it('handles onClick properly', () => {
    const onClick = sinon.spy();
    const shallowMount = shallow(
      <EmergencyAlert.Link onClick={onClick}>Link</EmergencyAlert.Link>
    );
    const shallowMountForAnonFunc = shallow(
      <EmergencyAlert.Link>Link</EmergencyAlert.Link>
    );

    shallowMount
      .find('.nhsuk-global-alert-link__margin-override')
      .simulate('click');
    shallowMountForAnonFunc
      .find('.nhsuk-global-alert-link__margin-override')
      .simulate('click');
    sinon.assert.called(onClick);
  });
});
