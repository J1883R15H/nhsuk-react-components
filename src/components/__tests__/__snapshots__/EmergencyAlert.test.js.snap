// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmergencyAlert all matches snapshot 1`] = `
<EmergencyAlert
  className=""
  id="nhsuk-global-alert"
  style={Object {}}
>
  <div
    className="nhsuk-global-alert "
    id="nhsuk-global-alert"
    style={Object {}}
  >
    <Container
      className=""
      fluid={false}
      style={Object {}}
    >
      <div
        className="nhsuk-width-container "
        style={Object {}}
      >
        <Row>
          <div
            className="nhsuk-grid-row"
          >
            <Column
              forceLayout={false}
              size="full"
              tabletWidth=""
            >
              <div
                className="nhsuk-grid-column-undefined  "
              >
                <div
                  className="nhsuk-global-alert__content"
                >
                  <Title
                    className=""
                    style={Object {}}
                    visuallyHiddenText="Alert: "
                  >
                    <h2
                      className="nhsuk-global-alert__heading "
                      style={Object {}}
                    >
                      <span
                        className="nhsuk-u-visually-hidden"
                      >
                        Alert: 
                      </span>
                      National Flu Outbreak
                    </h2>
                  </Title>
                  <Message
                    className=""
                    style={Object {}}
                  >
                    <p
                      className="nhsuk-global-alert__message "
                      style={Object {}}
                    >
                      There has been a national flu outbreak.
                      <Link
                        className=""
                        href="https://www.nhs.uk"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <a
                          className="nhsuk-u-nowrap nhsuk-global-alert-link__margin-override "
                          href="https://www.nhs.uk"
                          onClick={[Function]}
                          style={Object {}}
                        >
                          How does it affect me
                        </a>
                      </Link>
                    </p>
                  </Message>
                  <LastUpdated
                    className=""
                    style={Object {}}
                  >
                    <p
                      className="nhsuk-global-alert__updated "
                      style={Object {}}
                    >
                      Updated 23 mins ago
                    </p>
                  </LastUpdated>
                </div>
              </div>
            </Column>
          </div>
        </Row>
      </div>
    </Container>
  </div>
</EmergencyAlert>
`;

exports[`EmergencyAlert last updated matches snapshot 1`] = `
<p
  className="nhsuk-global-alert__updated "
  style={Object {}}
>
  Updated 23 mins ago
</p>
`;

exports[`EmergencyAlert link matches snapshot 1`] = `
<a
  className="nhsuk-u-nowrap nhsuk-global-alert-link__margin-override "
  href="#"
  onClick={[Function]}
  style={Object {}}
>
  Link
</a>
`;

exports[`EmergencyAlert message matches snapshot 1`] = `
<p
  className="nhsuk-global-alert__message "
  style={Object {}}
>
  Message
</p>
`;

exports[`EmergencyAlert title matches snapshot 1`] = `
<h2
  className="nhsuk-global-alert__heading "
  style={Object {}}
>
  <span
    className="nhsuk-u-visually-hidden"
  >
    Alert: 
  </span>
  Title
</h2>
`;
